internal/venice-common/src/main/java/com/linkedin/venice/pushstatushelper/PushStatusStoreWriter.java

```diff
@@ -119,49 +120,76 @@ public void addToSupposedlyOngoingIncrementalPushVersions(
       String incrementalPushVersion,
       ExecutionStatus status) {
     PushStatusKey pushStatusKey = PushStatusStoreUtils.getOngoingIncrementalPushStatusesKey(storeVersion);
-    PushStatusValueWriteOpRecord writeComputeRecord = new PushStatusValueWriteOpRecord();
-    instancesMapOps incrementalPushes = new instancesMapOps();
-    incrementalPushes.mapUnion = Collections.singletonMap(incrementalPushVersion, status.getValue());
-    incrementalPushes.mapDiff = Collections.emptyList();
-    writeComputeRecord.instances = incrementalPushes;
-    writeComputeRecord.reportTimestamp = new NoOp();
+    UpdateBuilder updateBuilder = new UpdateBuilderImpl(updateSchema);
+    updateBuilder
+        .setEntriesToAddToMapField("instances", Collections.singletonMap(incrementalPushVersion, status.getValue()));
     LOGGER.info(
         "Adding incremental push version: {} to ongoingIncrementalPushes of store: {} from instance: {}",
         incrementalPushVersion,
         storeName,
         instanceName);
     veniceWriterCache.prepareVeniceWriter(storeName)
-        .update(
-            pushStatusKey,
-            writeComputeRecord,
-            AvroProtocolDefinition.PUSH_STATUS_SYSTEM_SCHEMA_STORE.getCurrentProtocolVersion(),
-            derivedSchemaId,
-            null);
+        .update(pushStatusKey, updateBuilder.build(), valueSchemaId, derivedSchemaId, null);
   }
 
   public void removeFromSupposedlyOngoingIncrementalPushVersions(
       String storeName,
       int storeVersion,
       String incrementalPushVersion) {
     PushStatusKey pushStatusKey = PushStatusStoreUtils.getOngoingIncrementalPushStatusesKey(storeVersion);
-    PushStatusValueWriteOpRecord writeComputeRecord = new PushStatusValueWriteOpRecord();
-    instancesMapOps incrementalPushes = new instancesMapOps();
-    incrementalPushes.mapUnion = Collections.emptyMap();
-    incrementalPushes.mapDiff = Collections.singletonList(incrementalPushVersion);
-    writeComputeRecord.instances = incrementalPushes;
-    writeComputeRecord.reportTimestamp = new NoOp();
+    UpdateBuilder updateBuilder = new UpdateBuilderImpl(updateSchema);
+    updateBuilder.setKeysToRemoveFromMapField("instances", Collections.singletonList(incrementalPushVersion));
     LOGGER.info(
         "Removing incremental push version: {} from ongoingIncrementalPushes of store: {} from instance: {}",
         incrementalPushVersion,
         storeName,
         instanceName);
     veniceWriterCache.prepareVeniceWriter(storeName)
-        .update(
-            pushStatusKey,
-            writeComputeRecord,
-            AvroProtocolDefinition.PUSH_STATUS_SYSTEM_SCHEMA_STORE.getCurrentProtocolVersion(),
-            derivedSchemaId,
-            null);
+        .update(pushStatusKey, updateBuilder.build(), valueSchemaId, derivedSchemaId, null);
```

Comment:
Wow, it's nice to see how much less verbose the `UpdateBuilder` API is compared to using the `SpecificRecord`!

Reply:
Yeah, we should use it everywhere whenever possible. There should be basically no special case for system store.